datafusion/physical-expr/src/intervals/interval_aritmetic.rs

```diff
@@ -1079,6 +1094,117 @@ mod tests {
         Interval::make(lower, upper, (false, false))
     }
 
+    #[test]
+    fn boolean_interval_test() -> Result<()> {
+        // Demonstrate / document how boolean Intervals work
+        #[derive(Debug)]
+        struct TestCase {
+            lower: bool,
+            upper: bool,
+            /// expected bounds when lower is open, upper is open
+            expected_open_open: (bool, bool),
+            /// expected bounds when lower is open, upper is closed
+            expected_open_closed: (bool, bool),
+            /// expected bounds when lower is closes, upper is open
+            expected_closed_open: (bool, bool),
+            // Not: closed/closed is the same as lower/upper
+        }
+
+        let cases = vec![
+            TestCase {
+                lower: false,
+                upper: false,
+                expected_open_open: (true, false), // whole range
```

Comment:
An open `false` lower bound is mapped according to rule 1.

An open `false` upper bound is not mapped, right?

Reply:
I believe this is what the tests show, and I did not change the behavior.  However,  I don't know if this is correct or not. Perhaps @berkaysynnada  / @metesynnada  have some insight

Reply:
The second output (expected_open_closed)) of this TestCase is (false, false] is the same as [true, false] which is an invalid interval. As I mentioned [there](https://github.com/apache/arrow-datafusion/pull/7885/files#r1367732159), I think these cases should give an error. Maybe we can wait until I propose the PR which I am working on and planning to submit in a few days. It will explain these initialization issues neatly.